app/assets/javascripts/discourse/app/models/category-list.js

```diff
@@ -9,83 +8,79 @@ import Topic from "discourse/models/topic";
 import { bind } from "discourse-common/utils/decorators";
 import I18n from "discourse-i18n";
 
-const MAX_CATEGORIES_LIMIT = 25;
-
 const CategoryList = ArrayProxy.extend({
   init() {
+    this.set("content", this.categories || []);
     this._super(...arguments);
-    this.set("content", []);
     this.set("page", 1);
+    this.set("fetchedLastPage", false);
   },
 
   @bind
   async loadMore() {
-    if (this.isLoading || this.lastPage) {
+    if (this.isLoading || this.fetchedLastPage) {
       return;
     }
 
     this.set("isLoading", true);
 
-    const data = { page: this.page + 1, limit: MAX_CATEGORIES_LIMIT };
-    if (this.parentCategory) {
-      data.parent_category_id = this.parentCategory.id;
-    }
+    const data = { page: this.page + 1 };
     const result = await ajax("/categories.json", { data });
-    this.set("page", data.page);
-
-    result.category_list.categories.forEach((c) => {
-      const record = Site.current().updateCategory(c);
-      this.categories.pushObject(record);
-    });
-
-    this.set("isLoading", false);
 
+    this.set("page", data.page);
     if (result.category_list.categories.length === 0) {
-      this.set("lastPage", true);
+      this.set("fetchedLastPage", true);
     }
+    this.set("isLoading", false);
 
     const newCategoryList = CategoryList.categoriesFrom(this.store, result);
-    this.categories.pushObjects(newCategoryList.categories);
+    newCategoryList.forEach((c) => this.categories.pushObject(c));
   },
 });
 
 CategoryList.reopenClass({
-  categoriesFrom(store, result) {
-    const categories = CategoryList.create({ store });
-    const list = Category.list();
```

Comment:
There is no reason to pass `list` as an argument to the other functions when it is globally accessible.